### PR TITLE
Angular service impersonation mnrva 692

### PR DESCRIPTION
**Pull request by pras7430**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-692

 ## Description:
impersonation token Service added to minerva admin project 
 ## Testing:

 ## Screenshots:
(if applicable)

### CODE REVIEW

**Comment by lovedeep92** [comment]
@pras7430 , it is developed but we can't test it right as the service part is off for now?

**Comment by pras7430** [comment]
> @pras7430 , it is developed but we can't test it right as the service part is off for some time?

@lovedeep92 right! but I have shown you all if we provide a  rax token in the header it was working.

File: projects/admin/src/app/tab/_features/monitors/pages/details/details.component.ts [hunk 1]
```diff
@@ -106,6 +106,10 @@ export class DetailsComponent implements OnInit {
 
     this.route.params.subscribe(params => {
       this.id = params['id'];
+      // this.imperToken.getImpersonationToken({test:'any'}).subscribe(token =>{
```
Comment (FarajiA) [inline projects/admin/src/app/tab/_features/monitors/pages/details/details.component.ts:109]: If this block isn't necessary here we can remove.

File: projects/admin/src/app/_model/impersonationModel.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+}
+interface Access {
+    token: Token;
+}
+
+export interface ImpersonationToken {
+    access?: Access;
+}
+
+export interface impUser {
```
Comment (FarajiA) [inline projects/admin/src/app/_model/impersonationModel.ts:15]: So this looks fine for now, something I'm not sure if I ever pointed out is the body of this user data can have different properties depending on the tenant. We can for example have a `tenant` property in place of `username` more on this here https://pages.github.rackspace.com/ServiceAPIContracts/global-auth-keystone-extensions/api-reference/impersonate-operations.html

In some cases we may only have access to the tenant ID.

**Review by FarajiA — APPROVED** [review]

File: projects/admin/src/app/_model/impersonationModel.ts (hunk 2 above)
Comment (pras7430) [reply projects/admin/src/app/_model/impersonationModel.ts:15]: 😁 we can update in it coming stories.

File: projects/admin/src/app/tab/_features/monitors/pages/details/details.component.ts (hunk 1 above)
Comment (pras7430) [reply projects/admin/src/app/tab/_features/monitors/pages/details/details.component.ts:109]: 👍🏻

**Pull request merged**